server_tests/test/clientTests.js

```diff
@@ -791,514 +736,468 @@ describe("RUST API TEST SUITE", function() {
+                    .then( function onSuccess (res) {
                         Comparison.bublPollOutput(res, true, {'id': commandId, 'fingerprint': fingerprint});
                         fingerprint = res.body.fingerprint;
                         return Q.delay(4000);
-                    }))
+                    }, shouldNotGetHere)
                     .then( function() {
                         return testClient.bublPoll(commandId, fingerprint, 4);
                     })
-                    .then( Comparison.catchExceptions(done, function(res) {
+                    .then( function onSuccess (res) {
                         Comparison.bublPollOutput(res, false, {'id': commandId, 'fingerprint': fingerprint});
                         return testClient.bublStop(commandId);
-                    }))
-                    .then( Comparison.catchExceptions(done, function(res) {
+                    }, shouldNotGetHere)
+                    .then( function onSuccess (res) {
                         Comparison.bublStopOutput(res);
-                    }));
+                    }, shouldNotGetHere);
                 }
             })
-            .then( Comparison.catchExceptions(done, function(res) {
+            .then( function onSuccess (res) {
                 Comparison.bublCaptureVideoOutput(res);
-                done();
-            }));
-        });           
+            }, shouldNotGetHere);
+        });
 
-        it("Expect missingParameter Error. /osc/commands/_bublPoll cannot get updates when no commandId is provided", function(done) {
-            testClient.bublPoll(undefined, '')
-            .then( Comparison.catchExceptions(done, function(res) {
-                Comparison.missingParameterError(res);
-                done();
-            }));
+        it("Expect missingParameter Error. /osc/commands/_bublPoll cannot get updates when no commandId is provided", function() {
+            return testClient.bublPoll(undefined, '')
+            .then( shouldNotGetHere, function onError (err) {
+                Comparison.missingParameterError(err);
+            });
         });
 
-        it("Expect missingParameter Error. /osc/commands/_bublPoll cannot get updates when no fingerprint is provided", function(done) {
+        it("Expect missingParameter Error. /osc/commands/_bublPoll cannot get updates when no fingerprint is provided", function() {
             this.timeout(timeoutValue);
             var stopped = false;
-            testClient.takePicture(sessionId, function(res) {
+            return testClient.takePicture(sessionId, function(res) {
                 if (!stopped) {
-                    testClient.bublPoll(res.body.id)
-                    .then( Comparison.catchExceptions(done, function(res) {
-                        Comparison.missingParameterError(res);
-                    }));
+                    return testClient.bublPoll(res.body.id)
```

Comment:
Returning a promise inside the status callback does nothing. Again, need to use some deferred scheme.